### ISSUE TITLE
ContextualMenu internal item role does not appear in dom

### ISSUE THREAD

**Issue by RanaKh1** [microsoft/fluentui#18638]
### Environment Information

- **Package version(s)**: @fluentui/react 8.17.2 and up (all versions)
- **Browser and OS versions**: all (tested on chrome)

### Describe the issue:

Although the item specified (in IContextualMenuProps) is defined with role='link', the role 'link' does not appear in dom and this affects the screen reader experience as Narrator does not announce that the item is a link.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/rana-khell/pen/KKWYEmW

const { ContextualMenuItemType, IContextualMenuProps, DefaultButton, ThemeProvider, initializeIcons } = window.FluentUIReact;
const { useConst } = window.FluentUIReactHooks;

// Initialize icons in case this example uses them
initializeIcons();

const ContextualMenuDefaultExample: React.FunctionComponent = () => {
  const menuProps = useConst<IContextualMenuProps>(() => ({
    shouldFocusOnMount: true,
    items: [
      { 
        key: 'linkWithTarget',
        text: 'Help page',
        href: 'http://bing.com',
        role: 'link',
        target: '_blank' },
    ],
  }));
  return <DefaultButton text="Click for ContextualMenu" menuProps={menuProps} />;
};

const ContextualMenuDefaultExampleWrapper = () => <ThemeProvider><ContextualMenuDefaultExample /></ThemeProvider>;
ReactDOM.render(<ContextualMenuDefaultExampleWrapper />, document.getElementById('content'))

#### Actual behavior:

Narrator is not announcing the role as 'Link' about the "Help page" control because the specified role 'link' did not affect the element inside the dom.

#### Expected behavior:

The specified role 'link' should affect the element inside the dom.
according to (MAS 4.1.2 – Name, Role, Value)

**Comment by gouttierre** [comment]
@RanaKh1 - Thanks for filing this issue with us and providing the details into the problem.
@smhigley - Would you be able to confirm if this is a regression, or if this behavior is an issue?

**Comment by smhigley** [comment]
@RanaKh1 we explicitly set the role on menuitems in the ContextualMenu, since the only allowed roles are `menuitem`, `menuitemcheckbox`, and `menuitemradio`. Links cannot exist inside a menu -- they're semantically invalid, and would cause issues with screen reader mode switching, which would then interfere with keyboard navigation.

Hope that helps!
